examples/developer-tools/src/controller.ts

```diff
@@ -236,4 +272,73 @@ export class Controller {
     });
     menuEl.replaceChildren(...newItems);
   }
+
+  /** Shows the file upload modal when the user selects that option from the load menu. */
+  private handleLoadFromBlockFactory() {
+    this.viewModel.toggleFileUploadModal(true);
+  }
+
+  /**
+   * Handles the drag event that occurs when a user drops a file onto the file upload
+   * drag-n-drop zone. Gets the first file of the type we want and attempts to load it
+   * into the block factory editor.
+   *
+   * @param e drop event
+   */
+  private handleFileDrop(e: DragEvent) {
+    // Prevent default behavior (Prevent file from being opened)
+    e.preventDefault();
+
+    this.viewModel.toggleFileUploadWarning(false);
+
+    this.viewModel.fileDropZone.classList.remove('isDragging');
+
+    if (!e.dataTransfer.items) return;
+
+    const firstItem = [...e.dataTransfer.items].find((item) => {
```

Comment:
Do we want to support uploading multiple items? My understanding is that right now if you wanted to migrate to the new tool, you'd have to download + upload each blocks individually.

Reply:
Yeah, my goal is one file at a time. Supporting multiple file upload makes it harder to give feedback on what happens if one of them is successful and one isn't for example, and you can only look at one block at a time. So I'd prefer to keep it this way so I don't have to design that UI.